### PR TITLE
Netappfiles update sdk with api version 2021 06 01

### PR DESCRIPTION
**Pull request by hildurhodd**
Minimum changes made to support the new sdk version
- Use TokenCredential with AzureProfile instead of ServiceClientCredentials
- Use NetAppFilesManager instead of AzureNetAppFilesManagementClientImpl
- Remove Observables and CompletableFuture since it is not supported any more.
- Updated readme file to describe new way to authenticate

## Purpose

* Update to latest SDK  version: 1.0.0-beta.5

## Does this introduce a breaking change?

```
[x] Yes
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[x] Other... Please describe: SDK Version update and changes to support that version
```

### CODE REVIEW

File: src/main/java/poolchange/sdk/sample/main.java [hunk 1]
```diff
@@ -89,19 +85,19 @@ public static void main( String[] args )
...
             NetAppAccountInner newAccount = new NetAppAccountInner();
             newAccount.withLocation(location);
 
             try
             {
-                anfAccount = await(Creation.createANFAccount(anfClient, resourceGroupName, anfAccountName, newAccount));
+                anfAccount = Creation.createANFAccount(manager.serviceClient(), resourceGroupName, anfAccountName, newAccount);
             }
-            catch (CloudException e)
+            catch (Exception e)
```
Comment (paulomarquesc) [inline src/main/java/poolchange/sdk/sample/main.java:98]: >Exception [](http://example.com/codeflow?start=19&length=9)

CloudException got replaced by AzureError (at least in Python), and should not be replaced by the broader Exception, can you please apply the equivalent for Java? Please do the same on all samples.

**Review by paulomarquesc — CHANGES_REQUESTED** [review]
:clock1:

**Review by AdelGhabboun — APPROVED** [review]
Looks good

**Review by paulomarquesc — APPROVED** [review]
:shipit:

**Pull request merged**